File: frontend/demo/component/badge/badge-icons.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/badge';
import '@vaadin/horizontal-layout';
import '@vaadin/icon';
import '@vaadin/icons';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('badge-icons')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-horizontal-layout theme="spacing">
        <vaadin-badge>
          <vaadin-icon icon="vaadin:clock" slot="icon"></vaadin-icon>
          Pending
        </vaadin-badge>
        <vaadin-badge theme="success">
          <vaadin-icon icon="vaadin:check" slot="icon"></vaadin-icon>
          Confirmed
        </vaadin-badge>
        <vaadin-badge theme="warning">
          <vaadin-icon icon="vaadin:warning" slot="icon"></vaadin-icon>
          Warning
        </vaadin-badge>
        <vaadin-badge theme="error">
          <vaadin-icon icon="vaadin:exclamation-circle-o" slot="icon"></vaadin-icon>
          Denied
        </vaadin-badge>
      </vaadin-horizontal-layout>
      <!-- end::snippet[] -->
    `;
  }
}

File: src/main/java/com/vaadin/demo/component/badge/BadgeIcons.java
package com.vaadin.demo.component.badge;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.badge.Badge;
import com.vaadin.flow.component.badge.BadgeVariant;
import com.vaadin.flow.component.icon.VaadinIcon;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.router.Route;

@Route("badge-icons")
public class BadgeIcons extends HorizontalLayout {

    public BadgeIcons() {
        // tag::snippet[]
        Badge pending = new Badge("Pending", VaadinIcon.CLOCK.create());

        Badge confirmed = new Badge("Confirmed", VaadinIcon.CHECK.create());
        confirmed.addThemeVariants(BadgeVariant.SUCCESS);

        Badge warning = new Badge("Warning", VaadinIcon.WARNING.create());
        warning.addThemeVariants(BadgeVariant.WARNING);

        Badge denied = new Badge("Denied",
                VaadinIcon.EXCLAMATION_CIRCLE_O.create());
        denied.addThemeVariants(BadgeVariant.ERROR);

        // end::snippet[]

        add(pending, confirmed, warning, denied);
    }

    public static class Exporter extends DemoExporter<BadgeIcons> { // hidden-source-line
    } // hidden-source-line
}
